libdnf/conf/ConfigMain.cpp

```diff
@@ -178,6 +178,7 @@ class ConfigMain::Impl {
     OptionBool cacheonly{false};
     OptionBool keepcache{false};
     OptionString logdir{"/var/log"};
+    OptionString varsdir{"/etc/dnf/vars"};
```

Comment:
We probably want this to be an `OptionStringList` similar to `reposdir`, for basically the same reasons that `reposdir` is. The default should be `{{"/etc/dnf/vars", "/etc/yum/vars"}}`.

Reply:
Is there a good way to represent the same thing in the old configuration, or will we just add /etc/yum/vars as a supported directory when the new configuration structures is used by default?

Reply:
The latter. With the YUM->DNF change, `reposdir` became a list instead of just a single string, so you can set multiple directories to search.

Reply:
The latter. With the YUM->DNF change, `reposdir` became a list instead of just a single string, so you can set multiple directories to search. We'll do the same here for `varsdir`.

Reply:
I've changed all the various parts to OptionStringList, and I'm using `OptionStringList varsdir{{"/etc/dnf/vars", "/etc/yum/vars"}};`, but it looks like OptionStringList doesn't have a correct constructor for this?

```
/home/fedora/libdnf/libdnf/conf/ConfigMain.cpp:181:64: error: conversion from ‘<brace-enclosed initializer list>’ to ‘libdnf::OptionStringList’ is ambiguous
     OptionStringList varsdir{{"/etc/dnf/vars", "/etc/yum/vars"}};
                                                                ^
In file included from /home/fedora/libdnf/libdnf/conf/ConfigMain.hpp:33,
                 from /home/fedora/libdnf/libdnf/conf/ConfigMain.cpp:21:
/home/fedora/libdnf/libdnf/conf/OptionStringList.hpp:37:5: note: candidate: ‘libdnf::OptionStringList::OptionStringList(const string&)’
     OptionStringList(const std::string & defaultValue);
     ^~~~~~~~~~~~~~~~
/home/fedora/libdnf/libdnf/conf/OptionStringList.hpp:35:5: note: candidate: ‘libdnf::OptionStringList::OptionStringList(const ValueType&)’
     OptionStringList(const ValueType & defaultValue);
     ^~~~~~~~~~~~~~~~
```

If I add a third value, similar to reposdir, it compiles fine. I'm not sure what needs changed.

Reply:
I'm not sure either, but let's go ahead for now with having a third directory: `/etc/yum.vars.d`. Set it as: `{{"/etc/dnf/vars", "/etc/yum/vars", "/etc/yum.vars.d"}}`. I suspect there's a missing implementation for handling only two default options instead of three, but the code for this looks funny to my eyes...